pkg/uploader/uploader.go

```diff
@@ -62,16 +65,17 @@ func (up *uploader) Upload(rootDir string, dstURL string) error {
 	u.Path = path.Join(u.Path, "/sap/ams/v1/bundles/SAP.tar.gz")
 	r, err := http.NewRequest(http.MethodPost, u.String(), buf)
 	if err != nil {
-		return fmt.Errorf("could not create bundle upload request %w", err)
+		return fmt.Errorf("could not create dcl upload request %w", err)
 	}
 	r.Header.Set("Content-Type", "application/gzip")
 	resp, err := up.client.Do(r)
 	if err != nil {
-		return fmt.Errorf("bundle upload request unsuccessful: %w", err)
+		return fmt.Errorf("dcl upload request unsuccessful: %w", err)
 	}
 	defer resp.Body.Close()
+	up.logResponse(resp)
 	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusCreated && resp.StatusCode != http.StatusNotModified {
-		return fmt.Errorf("unexpected response: status(%s) body(%s)", resp.Status, resp.Body)
+		return fmt.Errorf("dcl upload failed")
```

Comment:
Let's remove the duplicate check of the StatusCode. We can either move this logic into logResponse or call logResponse only in case of error.

Reply:
So the printer also returns an error? we need to have the staging failed anyhow

Reply:
Either printer returns an error or, perhaps even better, something like:
```
if  resp.StatusCode != http.StatusOK... {
  up.logResponse(resp)
  return fmt.Errorf("DCL upload failed")
 }
  up.log.Info("Base DCLs uploaded and compiled successfully")
```
This way, we avoid checking the status code twice.

Reply:
fixed